verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -91,7 +94,17 @@ static absl::Status PreprocessSingleFile(
   config.filter_branches = true;
   config.include_files = true;
   // config.expand_macros=1;
-  verilog::VerilogPreprocess preprocessor(config);
+
+  verilog::VerilogProject project(".", preprocessing_info.include_dirs);
+
+  FileOpener file_opener =
+      [&project](
+          absl::string_view filename) -> absl::StatusOr<absl::string_view> {
```

Comment:
something to experiment with: Maybe we can even leave out the `-> absl::StatusOr<absl::string_view>` part. Ideally, the compiler will be able to deduce the return type to look at the left hand side of the expression.

(though I would also not be surprised if it can't.)

Reply:
I thought the compiler would be able to deduce the type, but it couldn't, so I am leaving it as it is.